### PR TITLE
Java startup detection of missing modules in a module with examples

### LINKED ISSUE #3824: Java Startup Detection of Missing Modules in a Module with Examples
### Proposed title of article
Java Startup Detection of Missing Modules in a Module with Examples

### Introduction paragraph (2-3 paragraphs):
A Java module is a packaging mechanism that enables you to package a Java application or Java API as a separate Java module. A Java module is packaged as a modular JAR file. A Java module can specify which of the Java packages it contains that should be visible to other Java modules which use this module. A Java module must also specify which other Java modules is required to do their job.

### Key takeaways: 
By the end of this tutorial, the reader will understand:
- What is a Module
- How to Write Modular Code
- How to create Java Modules
- How to run a Java Module
- Java Module Benefits
### References: 
N/A

### PR DESCRIPTION
**Pull request by nmercy21**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3824 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by lalith1403** [comment]
plagiarism check done ✔️

**Comment by BrianaNzivu** [comment]
Hey @nmercy21 👋 

Welcome to Section - EngEd.
We are happy to have you here 🚀

Very good job on the article. It is a very interesting read 👏 .

Although, your article has quite a lot of grammatical mistakes. I want you to read through and correct it using [Grammarly](https://app.grammarly.com/).

Ping me when done 🚀

**Comment by nmercy21** [comment]
Hello, @BrianaNzivu I've done the changes you requested. Kindly check.

File: content/articles/java-startup-detection-of-missing-modules-in-a-module-with-examples/index.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+We start by creating a java project to house our java modules. Creating a code is to make organized source code, tests, and libraries that are portable, small, and reliable. We name our java project `JavaModule`.
+
+![project2](/engineering-education/java-startup-detection-of-missing-modules-in-a-module-with-examples/project2.png)
+
+- Create a new java module
+
+![module](/engineering-education/java-startup-detection-of-missing-modules-in-a-module-with-examples/module.png)
+
+Java modules should have a unique name and should only contain full stops. The module creates a default package that contains the module-info.java class. The module-info.class is used to declare the dependencies within our java module project.
```
Comment (BrianaNzivu) [inline content/articles/java-startup-detection-of-missing-modules-in-a-module-with-examples/index.md:72]: Clearly highlight steps on how to create  Java module

**Comment by nmercy21** [comment]
Hello @BrianaNzivu I wish to enquire about the progress of my article.

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
@nmercy21 
Well done - please review some of the edits I made during my final review.

You'll be able to see your live published article on the site (in about 30 mins).

To give your article a boost in page engagement - share it across your social site (LinkedIn/Twitter) and tag the EngEd program so we can also share it and give it some traction as well!
(we will try and do the same once we feature it).

If you have not done so already - consider adding a skill section to your author page. This will go a long way for potential future employers.

**Pull request merged**